core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -246,57 +249,67 @@ export class ContributorStatsComponent {
 
     if (response.question_contribution_stats.length > 0) {
       this.statsData.questionContribution = this
-        .createQuestionContributionStatsPage(
-          response.question_contribution_stats);
+        .createAllQuestionStatsRecords(
+          response.question_contribution_stats, true);
     }
 
     if (response.question_review_stats.length > 0) {
-      this.statsData.questionReview = this.createQuestionReviewStatsPage(
-        response.question_review_stats);
+      this.statsData.questionReview = this.createAllQuestionStatsRecords(
+        response.question_review_stats, false);
     }
   }
 
-  createTranslationContributionStatsPage(
-      stat: TranslationContributionBackendDict): StatsPage {
-    return {
-      data: [this.createTranslationContributionStat(stat)],
-      language: this.languageUtilService.getAudioLanguageDescription(
-        stat.language_code),
-      currentPageStartIndex: 0,
-      currentPageEndIndex: 5
-    };
-  }
+  createInitialTranslationStatsRecord(
+      stat: TranslationContributionBackendDict | TranslationReviewBackendDict,
+      isTranslationContribution: boolean
+  ): StatsPage {
+    const data = [];
+    if (isTranslationContribution) {
+      // We can confirm that if isTranslationContribution == true
+      // the type of stat object is TranslationContributionBackendDict.
+      data.push(this.createTranslationContributionStat(
+        stat as TranslationContributionBackendDict));
+    } else {
+      // We can confirm that if isTranslationContribution == false
+      // the type of stat object is TranslationReviewBackendDict.
+      data.push(this.createTranslationReviewStat(
+        stat as TranslationReviewBackendDict));
+    }
 
-  createTranslationReviewStatsPage(
-      stat: TranslationReviewBackendDict): StatsPage {
     return {
-      data: [this.createTranslationReviewStat(stat)],
+      data,
       language: this.languageUtilService.getAudioLanguageDescription(
         stat.language_code),
       currentPageStartIndex: 0,
-      currentPageEndIndex: 5
+      currentPageEndIndex: this.ITEMS_PER_PAGE
     };
   }
 
-  createQuestionContributionStatsPage(
-      stats: QuestionContributionBackendDict[]): StatsPage {
-    return {
-      data: stats.map((stat) => {
-        return this.createQuestionContributionStat(stat);
-      }),
-      currentPageStartIndex: 0,
-      currentPageEndIndex: 5
-    };
-  }
+  createAllQuestionStatsRecords(
```

Comment:
Same thing, keep it as two separate functions. It would be fine to do a single function if the interior was pretty similar, but as it stands your function is basically a big if/else block and there isn't much benefit in unifying in that case.